go/src/lib.rs

```diff
@@ -379,6 +394,70 @@ pub unsafe extern "C" fn command(
                 command_response.float_value = num;
                 Ok(Some(command_response))
             }
+            Value::Boolean(bool) => {
+                command_response.bool_value = bool;
+                Ok(Some(command_response))
+            }
+            Value::Array(array) => {
+                let len = array.len();
+                let mut vec_elements_len: Vec<c_long> = Vec::<c_long>::with_capacity(len);
+                let vec = array
+                    .iter()
+                    .map(|value| {
+                        if *value == Value::Nil {
+                            vec_elements_len.push(0);
+                            return std::ptr::null_mut();
+                        }
+                        let res = <String>::from_redis_value(value)
+                            .expect("Couldn't read string from RedisValue")
+                            .chars()
+                            .map(|b| b as c_char)
+                            .collect::<Vec<_>>();
+                        let (res_ptr, res_len) = convert_vec_to_pointer(res);
+                        vec_elements_len.push(res_len);
+                        res_ptr
+                    })
+                    .collect::<Vec<_>>();
+                let (vec_ptr, len) = convert_vec_to_pointer(vec);
+                let (vec_elements_len_ptr, _) = convert_vec_to_pointer(vec_elements_len);
+                command_response.array_value = vec_ptr;
+                command_response.array_value_len = len;
+                command_response.array_elements_len = vec_elements_len_ptr;
+                Ok(Some(command_response))
+            }
+            Value::Map(map) => {
+                let mut vec_elements_len: Vec<c_long> = Vec::<c_long>::with_capacity(map.len() * 2);
+
+                let result = map
+                    .into_iter()
+                    .flat_map(|(key, value)| {
+                        let key = <String>::from_redis_value(&key)
```

Comment:
These lines are repetitive, either move them into a helper method / struct or make a macro out of them